config/configuration_test.go

```diff
@@ -16,259 +16,126 @@ package config
+		"Clients.Notification": Client{Host: "localhost", Protocol: "http", Port: 48060},
+		"Clients.Logging": Client{Host: "localhost", Protocol: "http", Port: 48061},
 	},
 }
 
-func TestSetConfig(t *testing.T) {
+func TestGetConfig(t *testing.T) {
 	tests := []struct {
 		name              string
 		env               Environment
 		configFilePath    string
-		configDirPath     string
+		result            Configuration
 		expectError       bool
-		expectedErrorType error
+		//expectedErrorType  *os.PathError
 	}{
 		{
-			name:              "Error ReadInConfig",
-			env:               createMockEnvReadInConfigErr(),
-			configFilePath:    TestConfigFilePath,
-			configDirPath:     TestConfigDirPath,
-			expectError:       true,
-			expectedErrorType: Error,
+			name:              "Successful GetDefaultConfig",
+			env:               getDefaultConfigFileMockEnvSuccess(),
+			configFilePath:    DefaultConfigFile,
+			expectError:       false,
+			//expectedErrorType: nil,
 		},
 		{
-			name:              "Successful SetConfig",
-			env:               createMockEnvSuccess(),
-			configFilePath:    TestConfigFilePath,
-			configDirPath:     TestConfigDirPath,
+			name:              "Successful GetConfig",
+			env:               getConfigFileMockEnvSuccess(),
+			configFilePath:    ValidConfigFile,
 			expectError:       false,
-			expectedErrorType: nil,
+			//expectedErrorType: nil,
 		},
-
 		{
-			name:              "Error Unmarshal",
-			env:               createMockEnvUnmarshalErr(),
-			configFilePath:    TestConfigFilePath,
-			configDirPath:     TestConfigDirPath,
+			name:              "Unsuccessful decode Config",
+			env:               getConfigFileMockDecodeError(),
+			configFilePath:    InvalidTomlConfigFile,
 			expectError:       true,
-			expectedErrorType: Error,
+			//expectedErrorType: toml.parseError,
 		},
 		{
-			name:              "Error WriteConfig",
-			env:               createMockEnvWriteConfigErr(),
-			configFilePath:    TestConfigFilePath,
-			configDirPath:     TestConfigDirPath,
+			name:              "Unsuccessful GetConfig",
+			env:               getConfigFileMockEnvError(),
+			configFilePath:    NonExistentConfigFile,
 			expectError:       true,
-			expectedErrorType: Error,
+			//expectedErrorType: &os.PathError{"stat", NonExistentConfigFile, syscall.Errno(2)},
 		},
 	}
+
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			err := SetConfig(test.env, test.configDirPath, test.configFilePath)
-
+			var err = LoadConfig(test.env)
+			fmt.Println(err)
 			if test.expectError && err == nil {
 				t.Error("We expected an error but did not get one")
 			}
 
 			if !test.expectError && err != nil {
-				t.Errorf("We do not expected an error but got one. %s", err.Error())
+				t.Errorf("We did not expect an error but got one. %s", err.Error())
 			}
 
-			if test.expectError {
+			// TODO fix error type checking
+			/*if test.expectError {
```

Comment:
toml.Decode function could return different kind of errors like - file does not exists, permissions denied or unable to parse the configuration.toml file because of wrong format and etc. Having all of them actually will not increase our test coverage. Our test coverage increase only if we have one test, but does not increase further if we have more than one test. Probably testing toml.decode by ourselves is not a good approach. Why should  we test this library? 
So, currently we have 60% test coverage of /config: `    github.com/edgexfoundry-holding/edgex-cli/config        0.003s  coverage: 60.0% of statements`.
My suggestion is to remove "Unsuccessful decode Config" test because toml.parseError is not exposed error type and stay just with the test of None existing file and uncomment all the code for checking err type. 

After executing "make test"
execute " go tool cover -html=coverage.out" shows exactly what code is tested and what is not.

Reply:
In addition all these tests were written because `Environment` interface used to have much more methods. Testing against mocked interface is much easier. But now actually we introduce `viper` only because of the tests ..that confuse me.  I am still considering if it worths